### ISSUE TITLE
Is it possible to float windows that don't have application IDs?

### ISSUE THREAD

**Issue by timvisher** [ianyh/Amethyst#795]
## System

 - **0.12.2 (66)**:
 - **10.14.3**:

## What's the problem?

I can't figure out a way to mark [dwarf fortress](http://www.bay12games.com/dwarves/) windows as floating by default. Presumably the only way to float windows by default is by adding it to the list of application IDs of which I don't think dwarf fortress has one.

## How can it be reproduced?

Launch dwarf fortress.

## What applications are involved?

Dwarf Fortress

## Has anything fixed it, even temporarily?

No

## dotfile (if you have one)

N/A

## Screenshots

N/A

**Comment by ianyh** [comment]
This came up in discussion in #748, but not currently no. I think the best way would be to make the wildcard apply to empty strings, and then use titles for specific apps.

**Comment by timvisher** [comment]
Maybe another question. I have always used the following script to get the id of a particular app and if it fails I assume the app has none.

```
#!/usr/bin/env bash

set -eu

set -x

path="${1:-}"

path="$(basename "$path")"

path="${path##.*}"

osascript -e 'id of app "'"$path"'"' | perl -ne 'chomp and print'
```

Is that your understanding of how to get the id of an app? Is there some other way? The windows I'm referring to are not 'created by a terminal'. Dwarf Fortress shows up in the Dock, has a menu bar with an application title, etc.

**Comment by dapicester** [comment]
I've been able to workaround this by adding a floating rule for empty bundleId and empty window title:

![Screen Shot 2020-05-22 at 11 51 26](https://user-images.githubusercontent.com/3007132/82629524-94f31080-9c22-11ea-86d9-9891f0342311.png)

**Comment by PlugFox** [comment]
Not working for me

**Comment by ianyh** [comment]
I think this is addressed by #1388?
